cmd/agent/main.go

```diff
@@ -519,11 +519,17 @@ func reconcileKubelet(ctx context.Context) (bool, error) {
 // delivered by Rancher if the generate_serving_certificate property
 // is set to 'true' for the clusters kubelet service.
 func KubeletNeedsNewCertificate(headers http.Header) error {
-	ipAddress := os.Getenv("CATTLE_ADDRESS")
 	currentHostname := os.Getenv("CATTLE_NODE_NAME")
-	fileSafeIPAddress := strings.ReplaceAll(ipAddress, ".", "-")
 
-	cert, err := tls.LoadX509KeyPair(fmt.Sprintf("/etc/kubernetes/ssl/kube-kubelet-%s.pem", fileSafeIPAddress), fmt.Sprintf("/etc/kubernetes/ssl/kube-kubelet-%s-key.pem", fileSafeIPAddress))
+	// RKE will save the certs on the node using either the public or private IP address, depending on the infrastructure provider.
+	// For example, the certs will be stored using the public IP address for VM's on digital ocean, but will use the private IP
```

Comment:
Tangentially related, do we know why RKE uses the public ip for digital ocean (or why AWS uses the private ip)? I'm assuming private is the correct option.

Reply:
DO only does so if it the private ip address option is enabled in UI, so that may also be related.

Reply:
I checked and it does seem to be the case that digital ocean clusters only report their external IP address in Rancher. Interestingly, for RKE2 clusters only the internal IP address is reported. I'll look to see if this is an existing issue or not, it seems odd though